Modules/_sqlite/cursor.c

```diff
@@ -550,13 +550,17 @@ _pysqlite_query_execute(pysqlite_Cursor* self, int multiple, PyObject* operation
                 if (!descriptor) {
                     goto error;
                 }
-                column_name = _pysqlite_build_column_name(self,
-                                sqlite3_column_name(self->statement->st, i));
-                if (!column_name) {
+                const char *colname = sqlite3_column_name(self->statement->st, i);
+                if (colname == NULL) {
+                    PyErr_NoMemory();
+                    goto error;
+                }
+                PyObject* colname_obj = _pysqlite_build_column_name(self, colname);
```

Comment:
`_pysqlite_build_column_name()` checks it argument for NULL. It is no longer needed.

Alternatively, you can call `PyErr_NoMemory()` in `_pysqlite_build_column_name()`.

Reply:
I think removing the check in `_pysqlite_build_column_name` is the better option.